src/manager.cpp

```diff
@@ -347,8 +347,100 @@ types::DbusVariantType
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    try
+    {
+        // Check if BMC reaches Ready state
+        const auto l_dbusValue = dbusUtility::readDbusProperty(
+            constants::bmcStateService, constants::bmcZeroStateObject,
+            constants::bmcStateInterface, constants::currentBMCStateProperty);
+
+        if (const auto l_currentBMCSate =
+                std::get_if<std::string>(&l_dbusValue))
+        {
+            if (*l_currentBMCSate != constants::bmcReadyState)
+            {
+                throw std::runtime_error(
+                    "BMC not ready. Single FRU VPD collection failed for " +
+                    std::string(i_dbusObjPath));
+            }
+        }
+
+        // Get system config JSON object from worker class
+        nlohmann::json l_sysCfgJsonObj{};
+
+        if (m_worker.get() != nullptr)
+        {
+            l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        // Check if system config JSON is present
+        if (l_sysCfgJsonObj.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON object not present. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // Get FRU path for the given D-bus object path from JSON
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_dbusObjPath);
+
+        if (l_fruPath.empty())
+        {
+            throw std::runtime_error(
+                "D-bus object path not present in JSON. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // Parse VPD
+        types::VPDMapVariant l_parsedVpd = m_worker->parseVpdFile(l_fruPath);
+
+        // If l_parsedVpd is pointing to std::monostate
+        if (l_parsedVpd.index() == 0)
+        {
+            throw std::runtime_error("VPD parsing failed for " +
+                                     std::string(i_dbusObjPath));
+        }
+
+        // Get D-bus object map from worker class
+        types::ObjectMap l_dbusObjectMap;
+        m_worker->populateDbus(l_parsedVpd, l_dbusObjectMap, l_fruPath);
+
+        if (l_dbusObjectMap.empty())
+        {
+            throw std::runtime_error(
+                "Failed to create D-bus object map. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // Call PIM's Notify method
+        if (!dbusUtility::callPIM(move(l_dbusObjectMap)))
+        {
+            throw std::runtime_error(
+                "Notify PIM failed. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // If PCA chip address is found in JSON, bind it.
+        const std::string& l_cmdToBindPCAChip =
+            jsonUtility::getPCAChipCmd(l_fruPath, l_sysCfgJsonObj, "bind");
```

Comment:
Why should JSON utiity provide this information.
If it is directly under a tag, it is ok to get from JSON, but Json utility should not be creating the command.